src/loaders/genome_collection/parse_tool_results.py

```diff
@@ -412,11 +439,15 @@ def _append_cell(
     else:
         raise ValueError(f'Unknown trait type {trait_type}')
 
-    cell = Cell(cell_id=str(cell_count), col_id=str(trait_idx), val=trait_val)
+    cell_id = str(cell_count)
+    cell = Cell(cell_id=cell_id, col_id=str(trait_idx), val=trait_val)
     heatmap_row.cells.append(cell)
 
+    cell_detail = _create_cell_detail(cell_id, detected_genes_score)
+
     return (min(min_value, trait_val),
-            max(max_value, trait_val)) if isinstance(trait_val, int) else (min_value, max_value)
+            max(max_value, trait_val),
+            cell_detail) if isinstance(trait_val, int) else (min_value, max_value, cell_detail)
 
```

Comment:
This is getting pretty hard to read. A regular if block would probably be more legible

Reply:
👍

Reply:
👍

Reply:
`Number` might be better than the tuple: https://docs.python.org/3/library/numbers.html#numbers.Number

Reply:
👍

Reply:
:+1: